extensions/src/main/java/org/pentaho/platform/web/servlet/PentahoXmlaServlet.java

```diff
@@ -373,11 +342,29 @@ public OlapConnection getConnection(
     };
   }
 
-  @Override public void init( ServletConfig servletConfig ) throws ServletException {
+  @Override
+  public void init( ServletConfig servletConfig ) throws ServletException {
     super.init( servletConfig );
-
     catalogLocator = makeCatalogLocator( servletConfig );
+  }
+
+  @VisibleForTesting
+  protected void isXMLAEnabled( OlapConnection connection ) {
```

Comment:
Do you think we could revise this method just to improve readability and really focus on its purpose? 
Meaning that:
- if the `unwrap` fails and goes to the catch, there is a log indeed but for the code it will assume that XMLA is enabled which seems odd;
- the body of the method is nested which makes it harder to read. Because of the name of the method I was expecting a return of some boolean which helped the "initial confusion".

I would suggest the following approach:
- try-catch only the unwrap and (maybe) throw a proper exception or if that is not required somehow, simply return and add a comment why.
- after that, remove the `if ( rolapConnection != null )` and let its content be the main code of that method.